src/toil/provisioners/__init__.py

```diff
@@ -81,6 +81,62 @@ def add_provisioner_options(parser: argparse.ArgumentParser) -> None:
                                            "Must be lowercase and may not contain the '_' character.")
 
 
+def parse_one_node_type(node_type_spec: Optional[str]) -> Optional[Tuple[Set[str], Optional[float]]]:
+    """
+    Similar to parse_node_types, except for <= 1 node type specifications.
+
+    Parse a specification for one (or zero) node types.
+
+    Takes a singular node types. The node type can have at least one instance type name (like 'm5a.large'
+    for AWS), and an optional bid in dollars after a colon.
+
+    Raises ValueError if the node type cannot be parsed.
+
+    Inputs should look something like this:
+
+    >>> parse_node_types('c5.4xlarge/c5a.4xlarge:0.42,t2.large')
+    ({'c5.4xlarge', 'c5a.4xlarge'}, 0.42)
+    >>> parse_node_types('t2.large')
+    ({'t2.large'}, None)
+
+    :param node_type_specs: A string defining a node type
+
+    :returns: the parsed node type, where the type is the set of
+              instance types, and the float bid, or None.
+    """
+    # Collect together all the node types
```

Comment:
I think a bunch of comments in here are also inaccurate now, if this is working just on one node type.